File: SocketOCDemo/SocketOCDemo/ViewController.m
//
//  ViewController.m
//  SocketOCDemo
//
//  Created by Vicent on 2022/2/20.
//

#import "ViewController.h"
#import "SocketOrignDemoViewController.h"
#import "CocoaAsyncSocketDemoViewController.h"

@interface ViewController ()

@property (nonatomic, strong) NSArray *dataSourceArray;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.dataSourceArray = @[@"SocketOrignDemo", @"CocoaAsyncSocketDemo"];
}

#pragma tableView--delegate
-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSourceArray.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *identify = @"CellIdentify";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identify];
    cell.textLabel.text = self.dataSourceArray[indexPath.row];
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    switch (indexPath.row) {
        case 0: {
            SocketOrignDemoViewController *vc = [[SocketOrignDemoViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
        }
            break;
        //CocoaAsyncSocketDemoViewController
        case 1: {
            CocoaAsyncSocketDemoViewController *vc = [[CocoaAsyncSocketDemoViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
        }
            break;
        default:
            break;
    }
}


@end

File: KeyPointSwiftDemo/KeyPointSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  KeyPointSwiftDemo
//
//  Created by Vicent on 2022/2/21.
//

import UIKit

class ViewController: UITableViewController {
    
    let dataSourceArray = ["ScrollViewUI--滚动视图UI", "原始约束", "VFL约束", "约束优先级", "TableView列表", "自定义视图", "闭包", "多线程", "响应式 链式 响应式 编程"]
    let vcArray: [UIViewController] = [ScrollDemoViewController(), OrignConstraintViewController(), VFLConstraintViewController(), ConstraintPriorityDemoViewController(), TableDemoViewController(), CustomViewDemoViewController(), ClosureDemoViewController(), ThreadDemoTableViewController(), ReactLinkFuncDemoViewController()]
    

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.dataSourceArray.count;
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cellIdentify: String = "CellID";
        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: cellIdentify)!;
        cell.textLabel?.text = self.dataSourceArray[indexPath.row];
        return cell;
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 50;
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let vc = self.vcArray[indexPath.row]
        vc.navigationItem.title = String(describing: type(of: self.vcArray[indexPath.row]))
        self.navigationController?.pushViewController(vc, animated: true)
    }


}

